src/ripple/app/transactors/SetSignerList.cpp

```diff
@@ -0,0 +1,322 @@
+
+    // Everything's ducky.  Add the ltSIGNER_LIST to the ledger.
+    SLE::pointer signerList (mEngine->entryCreate (ltSIGNER_LIST, index));
+    writeSignersToLedger (signerList, quorum, signers.vec);
+
+    // Lambda for call to dirAdd.
+    auto describer = [&] (SLE::ref sle, bool dummy)
+        {
+            Ledger::ownerDirDescriber (sle, dummy, mTxnAccountID);
+        };
+
+    // Add the signer list to the account's directory.
+    std::uint64_t hint;
+    TER result = mEngine->view ().dirAdd (
+        hint, getOwnerDirIndex (mTxnAccountID), index, describer);
+
+    if (m_journal.trace) m_journal.trace <<
+        "Create signer list for account " <<
+        mTxnAccountID << ": " << transHuman (result);
+
+    if (result != tesSUCCESS)
+        return result;
+
+    signerList->setFieldU64 (sfOwnerNode, hint);
+
+    // If we succeeded, the new entry counts against the creator's reserve.
+    mEngine->view ().increaseOwnerCount (mTxnAccount, addedOwnerCount);
+
+    return result;
+}
+
+TER SetSignerList::destroySignerList (uint256 const& index)
+{
+    // See if there's an ltSIGNER_LIST for this account.
+    SLE::pointer signerList =
+        mEngine->view ().entryCache (ltSIGNER_LIST, index);
+
+    // If the signer list doesn't exist we've already succeeded in deleting it.
+    if (!signerList)
+        return tesSUCCESS;
+
+    // We have to examine the current SignerList so we know how much to
+    // reduce the OwnerCount.
+    std::size_t removeFromOwnerCount = 0;
+    uint256 const signerListIndex = getSignerListIndex (mTxnAccountID);
+    SLE::pointer accountSignersList =
+        mEngine->view ().entryCache (ltSIGNER_LIST, signerListIndex);
+    if (accountSignersList)
+    {
+        STArray const& actualList =
+            accountSignersList->getFieldArray (sfSignerEntries);
+        removeFromOwnerCount = ownerCountDelta (actualList.size ());
+    }
+
+    // Remove the node from the account directory.
+    std::uint64_t const hint (signerList->getFieldU64 (sfOwnerNode));
+
+    TER const result  = mEngine->view ().dirDelete (false, hint,
+        getOwnerDirIndex (mTxnAccountID), index, false, (hint == 0));
+
+    if (result == tesSUCCESS)
+        mEngine->view ().decreaseOwnerCount (mTxnAccount, removeFromOwnerCount);
+
+    mEngine->view ().entryDelete (signerList);
+
+    return result;
+}
+
+TER SetSignerList::validateQuorumAndSignerEntries (
+    std::uint32_t quorum,
+    std::vector<SignerEntries::SignerEntry>& signers) const
+{
+    // Reject if there are too many or too few entries in the list.
+    {
+        std::size_t const signerCount = signers.size ();
+        if ((signerCount < SignerEntries::minEntries)
+            || (signerCount > SignerEntries::maxEntries))
+        {
+            if (m_journal.trace) m_journal.trace <<
+                "Too many or too few signers in signer list.";
+            return temMALFORMED;
+        }
+    }
+
+    // Make sure there are no duplicate signers.
```

Comment:
Do we not want to check that each entry in the SignerEntries is ordered by sfAccount ?  i.e. check that it's canonical?